src/tensor_operations/matrix_decomposition.jl

```diff
@@ -662,8 +662,9 @@ function factorize(A::ITensor, Linds...; kwargs...)
   # so eigen should only be used if a larger cutoff is requested)
   automatic_cutoff = 1e-12
   Lis = indices(Linds...)
-  dL, dR = dim(Lis), dim(indices(setdiff(inds(A), Lis)))
-  maxdim = get(kwargs, :maxdim, min(dL, dR))
+  dL, dR = dim(commoninds(inds(A), Lis)), dim(indices(setdiff(inds(A), Lis)))
+  # maxdim is forced to be at most the max given SVD
+  maxdim = min(get(kwargs, :maxdim, min(dL, dR)), min(dL, dR))
```

Comment:
Would you mind moving `maxdim` to the `factorize` function signature, say with `maxdim=nothing` to set a default of `nothing`? We are moving towards that style instead of using `get`. You'll have to then pass `maxdim` explicitly to `factorize_svd` and `factorize_eigen`.